go/store/nbs/journal_index_record.go

```diff
@@ -167,12 +167,16 @@ func readJournalIndexRecord(buf []byte) (rec indexRec, err error) {
 	return
 }
 
-func validateIndexRecord(buf []byte) (ok bool) {
-	if len(buf) > (indexRecLenSz + indexRecChecksumSz) {
-		off := len(buf) - indexRecChecksumSz
```

Comment:
previously, we assumed that `buf` was pre-sliced to the correct record length (it wasn't)